pipeline-model-definition/src/test/resources/postStage/catchErrorStageAborted.groovy

```diff
@@ -0,0 +1,56 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+pipeline {
+    agent any
+
+    stages {
+        stage("foo") {
+            steps {
+                catchError(buildResult: "SUCCESS",
+                    message: "Caught an error",
+                    stageResult: "ABORTED") {
+                    error("uhoh")
+                }
+            }
+            post {
+                success {
+                    echo "This shouldn't happen"
+                }
+                aborted {
+                    echo "This should happen"
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            echo "The build should be a success"
+        }
+        aborted {
+            echo "The build shouldn't be aborted"
```

Comment:
I'm actually a little surprised that this isn't happening (even though I think it's good that it isn't). Why don't we find the `WarningAction` for the nested stage?

In Blue Ocean this doesn't really matter because the parents of nested stages don't have their actual status reported, but I would've thought that in `isInGivenStage` here you would need to find nodes where the only enclosing block is the current stage, not where any of the enclosing blocks is the current stage.

Reply:
So `ABORTED` shouldn't be the overall build status, but there definitely should be testing for nested stages, which I don't have here.

Reply:
Ok, would be good to check to make sure, because if it does use the `WarningAction` from nested stages you'll need to tweak `findWorstWarningBetween` for your use case anyway so no point in making it public over in pipeline-graph-analysis.

Reply:
Yeah, I added a nested test. For now, I decided to just stick with the logic used elsewhere and treat parent stages with a child stage with a `WarningAction` as if the `WarningAction` was directly on the parent stage. I'm honestly not sure what the ideal behavior there _should_ be, but this feels fine for now.